core/lib/spree/core/importer/order.rb

```diff
@@ -4,47 +4,46 @@ module Importer
-            create_shipments_from_params(params.delete(:shipments_attributes), order)
-            create_adjustments_from_params(params.delete(:adjustments_attributes), order)
-            create_payments_from_params(params.delete(:payments_attributes), order)
-
-            if completed_at = params.delete(:completed_at)
-              order.completed_at = completed_at
-              order.state = 'complete'
-            end
+          ensure_country_id_from_params params[:ship_address_attributes]
+          ensure_state_id_from_params params[:ship_address_attributes]
+          ensure_country_id_from_params params[:bill_address_attributes]
+          ensure_state_id_from_params params[:bill_address_attributes]
+
+          create_params = params.slice :currency
+          order = Spree::Order.create! create_params
+          order.associate_user!(user)
+
+          shipments_attrs = params.delete(:shipments_attributes)
+
+          create_shipments_from_params(shipments_attrs, order)
+          create_line_items_from_params(params.delete(:line_items_attributes), order)
+          create_shipments_from_params(params.delete(:shipments_attributes), order)
+          create_adjustments_from_params(params.delete(:adjustments_attributes), order)
+          create_payments_from_params(params.delete(:payments_attributes), order)
+
+          if completed_at = params.delete(:completed_at)
+            order.completed_at = completed_at
+            order.state = 'complete'
+          end
 
-            params.delete(:user_id) unless user.try(:has_spree_role?, "admin") && params.key?(:user_id)
+          params.delete :user_id unless user.try(:has_spree_role?, "admin") && params.key?(:user_id)
 
-            order.update_attributes!(params)
+          order.update_attributes!(params)
 
-            order.create_proposed_shipments unless shipments_attrs.present?
+          order.create_proposed_shipments unless shipments_attrs.present?
 
-            # Really ensure that the order totals & states are correct
-            order.updater.update
-            if shipments_attrs.present?
-              order.shipments.each_with_index do |shipment, index|
-                shipment.update_columns(cost: shipments_attrs[index][:cost].to_f) if shipments_attrs[index][:cost].present?
-              end
+          # Really ensure that the order totals & states are correct
+          order.updater.update
+          if shipments_attrs.present?
+            order.shipments.each_with_index do |shipment, index|
+              shipment_cost = shipments_attrs[index][:cost]
+              shipment.update_columns(cost: shipment_cost.to_f) if shipment_cost.present?
             end
-            order.reload
-          rescue Exception => e
-            order.destroy if order && order.persisted?
-            raise e.message
           end
+          order.reload
+        rescue Exception => error
```

Comment:
Avoid rescuing the `Exception` class. Perhaps you meant to rescue `StandardError`?